src/cnxml/jsx.ts

```diff
@@ -0,0 +1,472 @@
+// Copyright 2020 OpenStax Poland
+// Licensed under the MIT license. See LICENSE file in the project root for
+// full license text.
+
+import { Case } from '../interfaces'
+
+/** Qualified name of an element or attribute */
+export interface Name {
+    /**
+     * Namespace
+     *
+     * When omitted for an element name it is inherited form the parent element.
+     * When omitted for an attribute, the name refers to the default namespace.
+     */
+    namespace?: string
+    /** Local name */
+    local: string
+}
+
+/** An element */
+export interface Element {
+    /** Element's name */
+    name: Name
+    /** Attributes set on this element */
+    attributes: Omit<JSX.IntrinsicAttributes, 'children'> & { [key: string]: any }
+    /** Children nodes */
+    children: Node
+}
+
+/** Any value that can be used as child of a JSX element */
+export type Node = Element | string | Node[]
+
+/** JSX element creator */
+export function createElement<
+    K extends keyof JSX.IntrinsicElements,
+    A extends JSX.IntrinsicAttributes,
```

Comment:
This will display an error in IDE for custom elements that are created by `adaptarr-front` like `highlight`, `insert`, `delete`, etc.